include/deal.II/lac/precondition.h

```diff
@@ -1925,7 +1924,8 @@ namespace internal
           // Check if we can initialize from vector that then gets set to zero
           // as the matrix will own the memory
           preconditioner->reinit(diagonal_inverse);
-          diagonal_inverse.reinit(0);
+          VectorType empty_vector;
```

Comment:
to make more obvious that you are not using this vector anywhere, here and below i would do
```
{
    VectorType empty_vector;
    diagonal_inverse.reinit(empty_vector);
}
```